src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java

```diff
@@ -1290,4 +1290,32 @@ private Integer foo(String bar) {
           )
         );
     }
+
+    @Test
+    void newInstanceMethodCall() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class A {
+                  int seen = 0;
+                  String lower(String s) {
+                      seen++;
+                      return s.toLowerCase();
+                  }
+              }
+              """
+          ),
+          java(
+            """
+              import java.util.stream.Stream;
+              class B {
+                  void bar(Stream<String> stream) {
+                      stream.map(s -> new A().lower(s));
```

Comment:
This used to insert the following, but that would then share the instance across operations, inflating the `seen` counter above.
```
                      stream.map(new A()::lower);
```

Reply:
@timtebeek I believe the same problem applies if the select is a method invocation. Imagine having `s -> a().lower(s)` with
```java
    A a() {
        return new A();
    }
```

Reply:
Turned out that case was already covered.

Reply:
Thanks for the rework! Hadn't spotted that other one, but like how you've solved it now.